### PR TITLE
feat(ecs): Implement method in ECS cluster to retrieve task ARN

### LINKED ISSUE #26232: (aws-ecs): Give Cluster a method that provides task ARNs
### Describe the feature

ARNs for tasks in a Cluster are in the form `arn:{partition}:ecs:{region}:{accountId}:task/{clusterName}/{taskId}`. It would be useful to have a method that can form these ARNs for you.

The aws-s3 Bucket class provides prior art for this with its [arnForObjects](https://github.com/aws/aws-cdk/blob/41cb2883e637a429c9eeb30c48544b69dbc7b065/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L747-L758) method.

I guess this method would be called `arnForTasks`.

### Use Case

In order to grant my task role permissions to the task protection API. My workaround is

```
        cluster_task_arn = Stack.of(self).format_arn(
            service='ecs',
            resource='task',
            resource_name=f'{fargate_cluster.cluster_name}/*'
        )
        fargate_task_definition.add_to_task_role_policy(
            aws_iam.PolicyStatement(
                actions=["ecs:UpdateTaskProtection"],
                resources=[cluster_task_arn],
            )
        )
```

With this method, the call would be 

```
        fargate_task_definition.add_to_task_role_policy(
            aws_iam.PolicyStatement(
                actions=["ecs:UpdateTaskProtection"],
                resources=[fargate_cluster.arn_for_tasks('*')],
            )
        )
```

### Proposed Solution

I'm not sure what best practice for forming ARNs within the CDK is, but using `formatArn` as I do in my example above would work.

### Other Information

See also [my request for a grantTaskProtection method](https://github.com/aws/aws-cdk/issues/26233)

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.85.0

### Environment details (OS name and version, etc.)

Ubuntu (Windows Subsystem for Linux)

### PR DESCRIPTION
**Pull request by badmintoncryer**
This pull request introduces a new method within the Cluster class, designed to retrieve the Amazon Resource Names (ARNs) of tasks that are part of a given ECS cluster. 

Example of how to get task ARN

```ts
declare cluster: Cluster;
// arn:aws:ecs:{region}:{regionId}:task/{clusterName}/*
const taskArnPattern = cluster.arnForTasks("*");
```

Closes #26232

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-with-iam-statement.ts
Comment (kaizencc) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-with-iam-statement.ts:1]: I don't see how this integ test is related to the feature added.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster.ts
Comment (kaizencc) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster.ts:1]: same for this

**Review by kaizencc — CHANGES_REQUESTED** [review]
Hi @badmintoncryer, a few comments on this PR. 1 is related to how you've implemented the arn, and the second is that the integ tests don't seem related. If you are interested in adding additional ecs integ tests, I would love that, but that should be part of a separate PR.

File: packages/aws-cdk-lib/aws-ecs/lib/cluster.ts [hunk 1]
```diff
@@ -583,6 +583,16 @@ export class Cluster extends Resource implements ICluster {
...
   }
 
+  /**
+   * Returns an ARN that represents all tasks within the cluster that match
+   * the task pattern specified. To represent all tasks, specify ``"*"``.
+   *
+   * @param keyPattern Task id pattern
+   */
+  public arnForTasks(keyPattern: string): string {
+    return `arn:${Stack.of(this).partition}:ecs:${Stack.of(this).region}:${Stack.of(this).account}:task/${this.clusterName}/${keyPattern}`;
```
[the remark this answers is not in the document]
Comment (badmintoncryer) [reply packages/aws-cdk-lib/aws-ecs/lib/cluster.ts]: Thank you very much. I have learned a lot.

**Comment by badmintoncryer** [comment]
Exemption Request

@kaizencc 
Thank you for your review.
I switched to using the formatArn() function and removed unnecessary integration tests.

File: packages/aws-cdk-lib/aws-ecs/README.md [hunk 2]
```diff
@@ -1601,3 +1617,5 @@ taskDefinition.addContainer('TheContainer', {
   }],
 });
 ```
+
+## Retrieve the task ARN which
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-ecs/README.md:1621]:
```suggestion
```

**Review by kaizencc — APPROVED** [review]
Thanks @badmintoncryer!

**Pull request merged**